examples/tp/tm_high_order.py

```diff
@@ -0,0 +1,527 @@
+print "drop until the TM learns the new sequence (Fig 2). What will be the prediction of"
+print "the TM if presented with the sequence BC? This would depend on what character"
+print "anteceding B. This is an important feature of high-order sequences."
+print "-"*50
+print ""
+
+for t in range(10):
+  tm.reset()
+  for k in range(4):
+    tm.compute(set(seq2[k][:].nonzero()[0].tolist()), learn=True)
+    activeColumnsIndeces = [tm.columnForCell(i) for i in tm.getActiveCells()]
+    predictedColumnIndeces = [tm.columnForCell(i) for i in tm.getPredictiveCells()]
+    actColState = ['1' if i in activeColumnsIndeces else '0' for i in range(tm.numberOfColumns())]
+    actColStr = ("".join(actColState))
+    currCols = [1 if i in activeColumnsIndeces else 0 for i in range(tm.numberOfColumns())]
+    acc = accuracy(currCols, predCols)
+    x.append(ts)
+    y.append(acc)
+    ts += 1
+    predCols = [1 if i in predictedColumnIndeces else 0 for i in range(tm.numberOfColumns())]
+    predColState = ['1' if i in predictedColumnIndeces else '0' for i in range(tm.numberOfColumns())]
+    predColStr = ("".join(predColState))
+
+# In this figure you can see how the TM starts making good predictions for particular
+# characters (spikes in the plot). Then, it will get half of its predictions right, which
+# correspond to the times in which is presented with character C. After some time, it
+# will learn correctly the sequence XBCY, and predict its characters accordingly.
+plt.ylim([-0.1,1.1])
+plt.xlim([40,80])
+plt.plot(x, y)
+plt.xlabel("Timestep")
+plt.ylabel("Prediction Accuracy")
+plt.title("Fig. 2: TM learns new sequence XBCY")
+plt.savefig("figure_2")
+plt.close()
+
+print ""
+print "-"*50
+print "We will present again each of the characters individually to the TM, that is,"
+print "not within any of the two sequences. When presented with character A the TM"
+print "predicts B, B predicts C, but this time C outputs a simultaneous prediction of"
+print "both D and Y. In order to disambiguate, the TM would require to know if the"
+print "preceding characters were AB or XB. When presented with character X the TM"
+print "predicts B, whereas Y and D yield no prediction."
+print "-"*50
+print ""
+
+for k in range(6):
+  print "--- " + "ABCDXY"[k] + " ---"
+  tm.compute(set(seqT[k][:].nonzero()[0].tolist()), learn=False)
+  activeColumnsIndeces = [tm.columnForCell(i) for i in tm.getActiveCells()]
+  predictedColumnIndeces = [tm.columnForCell(i) for i in tm.getPredictiveCells()]
+  currCols = [1 if i in activeColumnsIndeces else 0 for i in range(tm.numberOfColumns())]
+  predCols = [1 if i in predictedColumnIndeces else 0 for i in range(tm.numberOfColumns())]
+  print("Active cols: " + str(np.nonzero(currCols)[0]))
+  print("Predicted cols: " + str(np.nonzero(predCols)[0]))
+  print ""
+
+# PART 3. Now we will present noisy inputs to the TM. We will add noise to the sequence XBCY
```

Comment:
It would be good to do three levels of noise: 30% noise (we should see almost no impact on accuracy), 50% noise (some impact), and 90% noise (lots of impact).